File: app/src/main/java/com/zftx/pm/di/AppComponent.java
package com.zftx.pm.di;

import android.app.Application;
import android.support.v4.app.Fragment;
import com.mrtan.data.NetModule;
import com.zftx.pm.App;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.DispatchingAndroidInjector;
import javax.inject.Singleton;

/**
 * @author mrtan on 1/17/18.
 */

@Singleton @Component(modules = {
    AndroidInjectionModule.class, NetModule.class, ActivityModule.class, ViewModelModule.class
}) public interface AppComponent {

  @Component.Builder
  interface Builder{
    @BindsInstance Builder application(Application app);

    AppComponent build();
  }

  void inject(App app);

  DispatchingAndroidInjector<Fragment> fragmentInjector();
}

File: app/src/main/java/com/mrtan/app/di/AppComponent.kt
package com.mrtan.app.di

import android.app.Application
import com.mrtan.app.App
import com.mrtan.common.inject.CoroutineDispatcherModule
import com.mrtan.data.NetModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

/**
 * @author mrtan on 1/17/18.
 */

@Singleton
@Component(
    modules = [
      AndroidInjectionModule::class,
      NetModule::class,
      ActivityModule::class,
      ViewModelModule::class,
      CoroutineDispatcherModule::class
    ]
)
interface AppComponent {

  @Component.Builder
  interface Builder {
    @BindsInstance fun application(app: Application): Builder

    fun build(): AppComponent
  }

  fun inject(app: App)
}
